src/librustdoc/clean/mod.rs

```diff
@@ -1700,7 +1700,14 @@ impl Clean<Item> for ty::VariantDef {
         let kind = match self.ctor_kind {
             CtorKind::Const => Variant::CLike,
             CtorKind::Fn => Variant::Tuple(
-                self.fields.iter().map(|f| cx.tcx.type_of(f.did).clean(cx)).collect(),
+                self.fields
+                    .iter()
+                    .map(|f| {
+                        let kind = StructFieldItem(cx.tcx.type_of(f.did).clean(cx));
+                        let what_rustc_thinks = Item::from_def_id_and_parts(f.did, None, kind, cx);
+                        Item { visibility: Visibility::Inherited, ..what_rustc_thinks }
```

Comment:
Can you add a comment that this is true because it's an enum variant? At first I got it confused with `struct S(usize)` and was confused why it was always public.

Reply:
Done!